server/src/main/java/org/spine3/server/aggregate/AggregatePartRepository.java

```diff
@@ -24,26 +24,32 @@
 
 import java.lang.reflect.Constructor;
 
-import static org.spine3.server.entity.Entities.createAggregatePartEntity;
-import static org.spine3.server.entity.Entities.getAggregatePartConstructor;
+import static org.spine3.server.aggregate.Aggregates.createAggregatePartEntity;
+import static org.spine3.server.aggregate.Aggregates.createAggregateRootEntity;
 
 /**
  * Common abstract base for repositories that manage {@code AggregatePart}s.
  *
  * @author Alexander Yevsyukov
  */
-public abstract class AggregatePartRepository
-        <I, A extends AggregatePart<I, ?, ?>, R extends AggregateRoot<I>>
+public abstract class AggregatePartRepository<I, A extends AggregatePart<I, ?, ?>>
         extends AggregateRepository<I, A> {
 
-    private final R root;
+    private final Class<AggregateRoot<I>> rootClass;
 
     /**
      * {@inheritDoc}
      */
-    protected AggregatePartRepository(BoundedContext boundedContext, R root) {
+    protected AggregatePartRepository(BoundedContext boundedContext) {
         super(boundedContext);
-        this.root = root;
+        rootClass = getAggregateRootClass();
+    }
+
+    private <R extends AggregateRoot<I>> Class<R> getAggregateRootClass() {
+        @SuppressWarnings("unchecked")
+        // It is OK because value returned with `getAggregateRootClass` is fully controlled.
+        final Class<R> rootClass = (Class<R>) Aggregates.getAggregateRootClass(getEntityClass());
```

Comment:
After some thought: that's not a good design decision. That's a "Munchausen" pulling himself out of the dirt using own hair.

Let's try to figure out the `root` class from something else: generic parameter, `AggregatePartRepository` constructor etc.